### PR TITLE
increase fallback timeout

### PR DESCRIPTION
**Pull request by lszucs**

### CODE REVIEW

File: actions/test.go [hunk 1]
```diff
@@ -249,9 +249,9 @@ func TestPost(c buffalo.Context) error {
 	if timeout := postTestrequestModel.TestSpecification.TestTimeout; timeout != "" {
```
Comment (trapacska) [inline actions/test.go]: this meant for both iOS and Android? because on iOS it is 1800 sec

File: actions/test.go [hunk 2]
```diff
@@ -234,24 +241,33 @@ func TestPost(c buffalo.Context) error {
 		return c.Render(http.StatusInternalServerError, r.String("Invalid request"))
 	}
 
+	testAndroid := false
 	if postTestrequestModel.EnvironmentMatrix.AndroidDeviceList != nil {
```
Comment (trapacska) [inline actions/test.go]: postTestrequestModel.EnvironmentMatrix is not checked for nil

File: actions/test.go [hunk 3]
```diff
@@ -234,24 +241,33 @@ func TestPost(c buffalo.Context) error {
 		return c.Render(http.StatusInternalServerError, r.String("Invalid request"))
 	}
 
+	testAndroid := false
```
Comment (trapacska) [inline actions/test.go]: please move these checks into a separate function, for example testPlatform(...)...

also it would be better to check if the given test spec is not nil in a switch case based on this and the api ref: 
https://github.com/googleapis/google-api-go-client/blob/master/testing/v1/testing-gen.go#L2995

for example: 
```
switch {
	case AndroidInstrumentationTest != nil,
		AndroidRoboTest != nil,
		AndroidTestLoop != nil:
		// android
	case IosXcTest != nil,
		IosTestLoop != nil:
		// ios
	}
```

File: actions/test.go [hunk 4]
```diff
@@ -234,24 +241,33 @@ func TestPost(c buffalo.Context) error {
...
+		testIos = len(postTestrequestModel.EnvironmentMatrix.IosDeviceList.IosDevices) > 0
 	}
 
 	if timeout := postTestrequestModel.TestSpecification.TestTimeout; timeout != "" {
 		secs, err := strconv.ParseFloat(strings.TrimSuffix(timeout, "s"), 32)
 		if err == nil {
-			if secs > 2700.0 {
-				logger.Warn(fmt.Sprintf("Incoming TestSpecification.TestTimeout '%s' from build '%s' exceeds limit of '2700s', overriding it to '2700s'", timeout, appSlug))
-				postTestrequestModel.TestSpecification.TestTimeout = "2700s"
+			if testAndroid && secs > androidMaxTimeoutSecs {
```
Comment (trapacska) [inline actions/test.go]: this two separated check should be organized to have a single parameterizable block

File: actions/test.go [hunk 5]
```diff
@@ -211,6 +218,44 @@ func TestGet(c buffalo.Context) error {
 	return c.Render(http.StatusOK, r.JSON(steps))
 }
 
+// ensureTimeout limits testing.TestMatrix.TestSpecification.TestTimeout and returns a warning if TestTimeout was higher than the allowed.
+func ensureTimeout(matrix *testing.TestMatrix, appSlug string) string {
```
Comment (trapacska) [inline actions/test.go]: what about error return value? it is less confusing when you use error return type when you have a mutator

File: actions/test.go [hunk 6]
```diff
@@ -211,6 +218,44 @@ func TestGet(c buffalo.Context) error {
...
+		spec.AndroidRoboTest != nil,
+		spec.AndroidTestLoop != nil:
+		maxTimeoutSecs = androidMaxTimeoutSecs
+	case spec.IosXcTest != nil:
+		maxTimeoutSecs = iosMaxTimeoutSecs
+	}
+
+	if maxTimeoutSecs > 0 && secs > float64(maxTimeoutSecs) {
+		spec.TestTimeout = fmt.Sprintf("%ds", maxTimeoutSecs)
+		return fmt.Sprintf("Incoming TestSpecification.TestTimeout '%s' from build '%s' exceeds limit of '%ds', overriding it to '%ds'", timeout, appSlug, maxTimeoutSecs, maxTimeoutSecs)
```
Comment (trapacska) [inline actions/test.go]: this error message should be one level back - in an outer level scope, for example in line 296
this will have an extra positive effect, you can drop appSlug argument from the function, which doesn't really makes sense to use in a function "ensureTimeout"

File: actions/test.go [hunk 7]
```diff
@@ -211,6 +218,44 @@ func TestGet(c buffalo.Context) error {
...
+		return ""
+	}
+
+	maxTimeoutSecs := 0
+	switch {
+	case spec.AndroidInstrumentationTest != nil,
+		spec.AndroidRoboTest != nil,
+		spec.AndroidTestLoop != nil:
+		maxTimeoutSecs = androidMaxTimeoutSecs
+	case spec.IosXcTest != nil:
```
Comment (trapacska) [inline actions/test.go]: https://github.com/googleapis/google-api-go-client/blob/master/testing/v1/testing-gen.go#L3014

this spec is an iOS spec also

File: actions/test.go [hunk 8]
```diff
@@ -246,14 +291,9 @@ func TestPost(c buffalo.Context) error {
...
 
-	if timeout := postTestrequestModel.TestSpecification.TestTimeout; timeout != "" {
-		secs, err := strconv.ParseFloat(strings.TrimSuffix(timeout, "s"), 32)
-		if err == nil {
-			if secs > 2700.0 {
-				logger.Warn(fmt.Sprintf("Incoming TestSpecification.TestTimeout '%s' from build '%s' exceeds limit of '2700s', overriding it to '2700s'", timeout, appSlug))
-				postTestrequestModel.TestSpecification.TestTimeout = "2700s"
-			}
-		}
+	warning := ensureTimeout(postTestrequestModel, appSlug)
```
Comment (trapacska) [inline actions/test.go]: please move this into the if

File: actions/test.go (hunk 7 above)
Comment (godrei) [reply actions/test.go]: the dependency version we use does not know this test type: https://github.com/bitrise-io/addons-firebase-testlab/blob/master/vendor/google.golang.org/api/testing/v1/testing-gen.go#L2682-L2748
